kythe/cxx/indexer/cxx/GraphObserver.h

```diff
@@ -72,6 +70,25 @@ class ProfileBlock {
   const char* SectionName;
 };
 
+class HashRecorder {
+ public:
+  virtual void RecordHash(const std::string& hash, absl::string_view web64hash,
+                          absl::string_view original) {}
+  virtual ~HashRecorder() {}
+};
+
+class FileHashRecorder : public HashRecorder {
+ public:
+  FileHashRecorder(const std::string& path);
```

Comment:
This constructor definitely wants to be `explicit` and probably wants to take `absl::string_view` (even if it will need to be converted to a null-terminated C-string to pass to fopen)

https://abseil.io/tips/1